app/Events/User/BaseUserEvent.php

```diff
@@ -7,7 +7,7 @@
 use Illuminate\Foundation\Events\Dispatchable;
 use Illuminate\Queue\SerializesModels;
 
-class Updated
+abstract class BaseUserEvent
```

Comment:
What is the benefit of the base event beyond the user class member?

Reply:
The base class eliminates the need to have the user variable constructor in each class. It essentially just DRYs update all events that take a user as a parameter.